src/services/user/hooks/userService.js

```diff
@@ -70,6 +70,37 @@ const checkUnique = (hook) => {
 	});
 };
 
+const checkUniqueEmail = async (hook) => {
+	// const userService = hook.service;
+	const userService = hook.app.service('users');
+
+	const { email } = hook.data;
+	if (!email) {
+		// there is no email address given. Nothing to check...
+		return Promise.resolve(hook);
+	}
+
+	// get userId of user entry to edit
+	const editUserId = hook.id;
+
+	const query = { email: email.toLowerCase() };
+
+	// check if new user or update of existing entry
+	if (editUserId) {
+		// exclude existing user entry
+		query._id = { $ne: editUserId };
+	}
+
+	return userService.find({ query }).then((result) => {
+		const users = (result || {}).data || [];
+		if (users.length === 0) {
+			return Promise.resolve(hook);
+		}
+
+		return Promise.reject(new BadRequest(`Die E-Mail Adresse ${email} ist bereits in Verwendung!`));
```

Comment:
Please check if it is possible to test which email exist over this way. Without jwt and with jwt.